examples/seq2seq/seq2seq.py

```diff
@@ -0,0 +1,302 @@
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--resume', '-r', default='',
+                        help='resume the training from snapshot')
+    parser.add_argument('--unit', '-u', type=int, default=1024,
+                        help='number of units')
+    parser.add_argument('--layer', '-l', type=int, default=3,
+                        help='number of layers')
+    parser.add_argument('--min-source-sentence', type=int, default=1,
+                        help='minimium length of source sentence')
+    parser.add_argument('--max-source-sentence', type=int, default=50,
+                        help='maximum length of source sentence')
+    parser.add_argument('--min-target-sentence', type=int, default=1,
+                        help='minimium length of target sentence')
+    parser.add_argument('--max-target-sentence', type=int, default=50,
+                        help='maximum length of target sentence')
+    parser.add_argument('--out', '-o', default='result',
+                        help='directory to output the result')
+    args = parser.parse_args()
+
+    source_ids = load_vocabulary(args.SOURCE_VOCAB)
+    target_ids = load_vocabulary(args.TARGET_VOCAB)
+    train_source = load_data(source_ids, args.SOURCE)
+    train_target = load_data(target_ids, args.TARGET)
+    assert len(train_source) == len(train_target)
+    train_data = [(s, t)
+                  for s, t in six.moves.zip(train_source, train_target)
+                  if args.min_source_sentence <= len(s)
+                  <= args.max_source_sentence and
+                  args.min_source_sentence <= len(t)
+                  <= args.max_source_sentence]
+    train_source_unknown = calculate_unknown_ratio(
+        [s for s, _ in train_data])
+    train_target_unknown = calculate_unknown_ratio(
+        [t for _, t in train_data])
+
+    print('Source vocabulary size: %d' % len(source_ids))
+    print('Target vocabulary size: %d' % len(target_ids))
+    print('Train data size: %d' % len(train_data))
+    print('Train source unknown ratio: %.2f%%' % (train_source_unknown * 100))
+    print('Train target unknown ratio: %.2f%%' % (train_target_unknown * 100))
+
+    target_words = {i: w for w, i in target_ids.items()}
+    source_words = {i: w for w, i in source_ids.items()}
+
+    model = Seq2seq(args.layer, len(source_ids), len(target_ids), args.unit)
+    if args.gpu >= 0:
+        chainer.cuda.get_device(args.gpu).use()
+        model.to_gpu(args.gpu)
+
+    optimizer = chainer.optimizers.Adam()
+    optimizer.setup(model)
+
+    train_iter = chainer.iterators.SerialIterator(train_data, args.batchsize)
+    updater = training.StandardUpdater(
+        train_iter, optimizer, converter=convert, device=args.gpu)
+    trainer = training.Trainer(updater, (args.epoch, 'epoch'))
+    trainer.extend(extensions.LogReport(trigger=(200, 'iteration')),
+                   trigger=(200, 'iteration'))
```

Comment:
Is it correct? This setting will not take average training loss over 200 iterations.